### ISSUE TITLE
Detected that custom integration 'irm_kmi' sets option flow config_entry explicitly, which is deprecated at custom_components/irm_kmi/config_flow.py, line 110

### ISSUE THREAD

**Issue by Doudy** [jdejaegh/irm-kmi-ha#62]
**Version**
 - Home Assistant: [e.g. 2024.12.5]
 - IRM KMI integration [e.g. 0.2.24]

Find in Log
Enregistreur: homeassistant.components.recorder.db_schema
Source: components/recorder/db_schema.py:624
intégration: Recorder (documentation, problèmes)
S'est produit pour la première fois: 28 décembre 2024 à 09:27:58 (88 occurrences)
Dernier enregistrement: 10:05:55

State attributes for weather.maison exceed maximum size of 16384 bytes. This can cause database performance issues; Attributes will not be stored

Enregistreur: homeassistant.helpers.frame
Source: helpers/frame.py:324
S'est produit pour la première fois: 10:07:50 (2 occurrences)
Dernier enregistrement: 10:08:59

Detected that custom integration 'irm_kmi' sets option flow config_entry explicitly, which is deprecated at custom_components/irm_kmi/config_flow.py, line 110: self.config_entry = config_entry. This will stop working in Home Assistant 2025.12, please create a bug report at https://github.com/jdejaegh/irm-kmi-ha/issues
Detected that custom integration 'irm_kmi' uses `async_config_entry_first_refresh`, which is only supported for coordinators with a config entry at custom_components/irm_kmi/__init__.py, line 27: await coordinator.async_config_entry_first_refresh(). This will stop working in Home Assistant 2025.11, please create a bug report at https://github.com/jdejaegh/irm-kmi-ha/issues
Enregistreur: homeassistant.helpers.frame
Source: helpers/frame.py:324
S'est produit pour la première fois: 10:07:50 (2 occurrences)
Dernier enregistrement: 10:08:59

Detected that custom integration 'irm_kmi' sets option flow config_entry explicitly, which is deprecated at custom_components/irm_kmi/config_flow.py, line 110: self.config_entry = config_entry. This will stop working in Home Assistant 2025.12, please create a bug report at https://github.com/jdejaegh/irm-kmi-ha/issues
Detected that custom integration 'irm_kmi' uses `async_config_entry_first_refresh`, which is only supported for coordinators with a config entry at custom_components/irm_kmi/__init__.py, line 27: await coordinator.async_config_entry_first_refresh(). This will stop working in Home Assistant 2025.11, please create a bug report at https://github.com/jdejaegh/irm-kmi-ha/issues

**Comment by jdejaegh** [comment]
Hello,

Thanks for the report!  I would help if you could describe what you were doing to get those lines to show up in the logs.  I managed to reproduce the two issues anyways. 

The first issue (`'irm_kmi' sets option flow config_entry explicitly`) is triggered when modifying the configuration of the integration.  It produces the following line:

```
2024-12-29 15:38:46.827 WARNING (MainThread) [homeassistant.helpers.frame] Detected that custom integration 'irm_kmi' sets option flow config_entry explicitly, which is deprecated at custom_components/irm_kmi/config_flow.py, line 110: self.config_entry = config_entry. This will stop working in Home Assistant 2025.12, please create a bug report at https://github.com/jdejaegh/irm-kmi-ha/issues
```

The second issue (`'irm_kmi' uses async_config_entry_first_refresh, which is only supported for coordinators with a config entry`) is triggered when using the repair flow.  It produces the following line:

```
2024-12-29 15:45:11.775 WARNING (MainThread) [homeassistant.helpers.frame] Detected that custom integration 'irm_kmi' uses `async_config_entry_first_refresh`, which is only supported for coordinators with a config entry at custom_components/irm_kmi/__init__.py, line 27: await coordinator.async_config_entry_first_refresh(). This will stop working in Home Assistant 2025.11, please create a bug report at https://github.com/jdejaegh/irm-kmi-ha/issues
```

I have to investigate further to see what the correct fix is.

**Comment by jdejaegh** [comment]
Both issues should be solved by PR #65 and #66.  The changes will be included in the next version of the integration.  Feel free to test and report any error you might encounter
